File: modules/stop/src/main/java/com/media/bus/stop/dto/external/SeoulBusStopRow.java
package com.media.bus.stop.dto.external;

import com.fasterxml.jackson.annotation.JsonProperty;

/// 서울 열린데이터광장 busStopLocationXyInfo API 단건 행(row) DTO
/// 참고: [...](http://openapi.seoul.go.kr:8088/){serviceKey}/json/busStopLocationXyInfo/{start}/{end}/
public record SeoulBusStopRow(

    @JsonProperty("STOPS_NO")
    String stopsNo,

    @JsonProperty("STOPS_NM")
    String stopsName,

    @JsonProperty("XCRD")
    String xCrd,

    @JsonProperty("YCRD")
    String yCrd,

    @JsonProperty("NODE_ID")
    String nodeId,

    @JsonProperty("STOPS_TYPE")
    String stopsType

) {}

File: modules/stop/src/main/kotlin/com/media/bus/stop/dto/external/SeoulBusStopRow.kt
package com.media.bus.stop.dto.external

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * ## 서울 열린데이터광장 busStopLocationXyInfo API 단건 행(row) DTO
 *
 * 참고: `http://openapi.seoul.go.kr:8088/{serviceKey}/json/busStopLocationXyInfo/{start}/{end}/`
 */
data class SeoulBusStopRow(
    @JsonProperty("STOPS_NO")
    val stopsNo: String,

    @JsonProperty("STOPS_NM")
    val stopsName: String,

    @JsonProperty("XCRD")
    val xCrd: String,

    @JsonProperty("YCRD")
    val yCrd: String,

    @JsonProperty("NODE_ID")
    val nodeId: String,

    @JsonProperty("STOPS_TYPE")
    val stopsType: String,
)
